File: DemoProject/app/src/main/java/com/ybdevelopers/demoproject/models/Feature.java

package com.ybdevelopers.demoproject.models;

import java.util.List;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Feature {

    @SerializedName("feature_id")
    @Expose
    private String featureId;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("options")
    @Expose
    private List<Option> options = null;

    public String getFeatureId() {
        return featureId;
    }

    public void setFeatureId(String featureId) {
        this.featureId = featureId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Option> getOptions() {
        return options;
    }

    public void setOptions(List<Option> options) {
        this.options = options;
    }

}

File: MobileFeature/app/src/main/java/com/ybdevelopers/mobilefeature/model/Feature.kt
package com.ybdevelopers.mobilefeature.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName


class Feature : Parcelable {
    @SerializedName("feature_id")
    @Expose
    var featureId: String? = null

    @SerializedName("name")
    @Expose
    var name: String? = null

    @SerializedName("options")
    @Expose
    var options: List<Option?>? = null

    protected constructor(`in`: Parcel) {
        featureId =
            `in`.readValue(String::class.java.classLoader) as String?
        name = `in`.readValue(String::class.java.classLoader) as String?
        `in`.readList(options!!, Option::class.java.classLoader)
    }

    constructor()

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(featureId)
        dest.writeValue(name)
        dest.writeList(options)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<Feature?> = object : Parcelable.Creator<Feature?> {
            override fun createFromParcel(`in`: Parcel): Feature? {
                return Feature(`in`)
            }

            override fun newArray(size: Int): Array<Feature?> {
                return arrayOfNulls(size)
            }
        }
    }
}